tests/integration/test_array.py

```diff
@@ -247,6 +247,19 @@ def main() -> tuple[array[int, 3], array[qubit, 2]]:
     validate(module.compile())
 
 
+def test_linear_for_loop(validate):
+    module = GuppyModule("test")
+    module.load(qubit, discard)
+
+    @guppy(module)
+    def main() -> None:
+        qs = array(qubit(), qubit(), qubit())
+        for q in qs:
+            discard(q)
+
+    validate(module.compile())
```

Comment:
I am just wondering here if you could check that the module *does* include the expect_none stuff, and then have a counterexample
```
qs = array(qubit(), qubit(), qubit())
[discard(q) for q in qs]
```
that does not need expect none (because the array returned by the comprehension is copyable)....I think??

Reply:
>  (because the array returned by the comprehension is copyable)....I think??

Yes, indeed! Unfortunately we can't add this test since #613 isn't merged yet